crates/core/src/lib.rs

```diff
@@ -48,3 +48,49 @@ pub mod file_group;
 pub mod storage;
 pub mod table;
 pub mod util;
+
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Config '{0}' not found")]
+    ConfNotFound(String),
+
+    #[error("Invalid config item '{item}', {source:?}")]
+    InvalidConf {
+        item: &'static str,
+        source: Box<dyn std::error::Error + Sync + Send + 'static>,
+    },
+
+    #[error("Parse url '{url}' failed, {source}")]
+    UrlParse {
+        url: String,
+        source: url::ParseError,
+    },
+
+    #[error("Invalid file path '{name}', {detail}")]
+    InvalidPath { name: String, detail: String },
+
+    #[error("{0}")]
```

Comment:
```suggestion
    #[error("Operation is not supported. {0}")]
```
Add more context

Reply:
this will be overwritten in a follow up.